src/utils/httpUtils.js

```diff
@@ -72,70 +131,90 @@ function doBulkUpsert(url, userToken, proxy, arr) {
   url += bulkUpsertEndpoint;
   return new Promise((resolve, reject) => {
     debug('Bulk upserting %d samples to %s', arr.length, url);
-    doPost(url, userToken, proxy, arr)
-    .end((err, res) => {
-      if (err) {
-        debug('doBulkUpsert err %O', err);
-        logger.error(err.message);
-        /*
-         * Don't Promise.reject(...) this error, because there is no handler
-         * for the rejection.
-         */
-        return resolve(err);
-      }
-
+    doPost(url, userToken, proxy, arr, intervalSecs)
+    .then((res) => {
       debug('doBulkUpsert returned OK %O', res.body);
       logger.info({
         activity: 'bulkUpsert',
         numSamples: arr.length,
       });
       return resolve(res);
+    },
+
+    (err) => {
+      debug('doBulkUpsert err %O', err);
+      logger.error(err.message);
+      /*
+       * Don't Promise.reject(...) this error, because there is no handler
+       * for the rejection.
+       */
+      return resolve(err);
     });
   });
 } // doBulkUpsert
 
 /**
  * Find Refocus subjects using query parameters.
  *
- * @param  {String} url - The Refocus url.
- * @param  {String} token - The Authorization token to use.
- * @param  {String} proxy - Optional proxy url
- * @param {String} qry - the query string
+ * @param {Object} generator - Generator object from the heartbeat. Contains:
+ *  {String} url - The Refocus url.
+ *  {String} token - The Authorization token to use.
+ *  {String} proxy - Optional proxy url
+ *  {String} subjectQuery - the query string
+ * @throws {ValidationError} if argument(s) is missing
```

Comment:
@returns -- doesn't this return the updated generator now?

Reply:
true. forgot to update